src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2094,7 +2114,46 @@ public void HandleIDebugEntryPointEvent2(IDebugEngine2 pEngine, IDebugProcess2 p
 
         public void HandleIDebugBreakpointEvent2(IDebugEngine2 pEngine, IDebugProcess2 pProcess, IDebugProgram2 pProgram, IDebugThread2 pThread, IDebugEvent2 pEvent)
         {
-            FireStoppedEvent(pThread, StoppedEvent.ReasonValue.Breakpoint);
+            bool fireStopEvent = true;
+
+            if (pEvent is IDebugBreakpointEvent2 debugBreakpointEvent) {
+                debugBreakpointEvent.EnumBreakpoints(out IEnumDebugBoundBreakpoints2 pBoundBreakpoints);
+                IDebugBoundBreakpoint2[] boundBp = new IDebugBoundBreakpoint2[1];
+                uint numReturned = 0;
+                while (pBoundBreakpoints.Next(1, boundBp, ref numReturned) == HRConstants.S_OK && numReturned == 1)
+                {
+                    if (boundBp[0].GetPendingBreakpoint(out IDebugPendingBreakpoint2 ppPendingBreakpoint) == HRConstants.S_OK)
+                    {
+                        if (ppPendingBreakpoint.GetBreakpointRequest(out IDebugBreakpointRequest2 ppBPRequest) == HRConstants.S_OK)
+                        {
+                            if (ppBPRequest is AD7BreakPointRequest ad7BreakpointRequest)
+                            {
+                                if (m_tracepointManager.Contains(ad7BreakpointRequest.Id))
+                                {
+                                    fireStopEvent = false;
```

Comment:
>fireStopEvent [](start = 36, length = 13)

We need to either break out of the enumeration if we find one, or change this code to find multiple breakpoints up front. As you have it, if we ever found multiple breakpoints you would continue the process multiple times.

Reply:
Is there a case where we would get multiple breakpoints where there are logpoints and regular breakpoints?

Reply:
In DAP, the only way it could happen is if you had both multiple types of breakpoint (ex: function breakpoint and line) on the same instruction.

Reply:
Actually, I guess that isn't 100% true, because you could have multiple line breakpoints that all wound up pointing at the same instruction...